app/ts/simulation/priceEstimator.ts

```diff
@@ -21,99 +22,96 @@ export class PriceEstimator {
+		const IUniswapV3Pool = new Interface(UniswapV3PairABI)
+		const IMulticall3 = new Interface(Multicall3ABI)
+
+		for (const token of tokens) {
+			if (token.address === quoteToken.address) {
+				tokenPrices.push({
+					token,
+					quoteToken,
+					price: 10n ** 18n
 				})
 				continue
 			}
-			const swapTransactions = [
-				{
-					type: '1559' as const,
-					from: sender,
-					chainId: chainId,
-					nonce: transactionCount,
-					maxFeePerGas: 0n,
-					maxPriorityFeePerGas: 0n,
-					gas: 15000000n,
-					to: UNISWAP_V2_ROUTER_ADDRESS,
-					value: 10n ** 18n,
-					input: stringToUint8Array(swapInterface.encodeFunctionData(
-						'swapExactETHForTokens',
-						[amountOutMin, [addressString(CHAINS[chainString].weth), addressString(token.token)], addressString(sender), deadline]
-					)),
-					accessList: [],
-				},
-				{
-					type: '1559' as const,
-					from: sender,
-					chainId: chainId,
-					nonce: transactionCount + 1n,
-					maxFeePerGas: 0n,
-					maxPriorityFeePerGas: 0n,
-					gas: 15000000n,
-					to: token.token,
-					value: 0n,
-					input: stringToUint8Array(swapInterface.encodeFunctionData('approve', [addressString(UNISWAP_V2_ROUTER_ADDRESS), 2n ** 127n] )),
-					accessList: [],
-				},
-				{
-					type: '1559' as const,
-					from: sender,
-					chainId: chainId,
-					nonce: transactionCount + 2n,
-					maxFeePerGas: 0n,
-					maxPriorityFeePerGas: 0n,
-					gas: 15000000n,
-					to: UNISWAP_V2_ROUTER_ADDRESS,
-					value: 0n,
-					input: stringToUint8Array(swapInterface.encodeFunctionData(
-						'swapTokensForExactETH',
-						[10n ** 18n / 2n, 2n ** 127n, [addressString(token.token), addressString(CHAINS[chainString].weth)], addressString(sender), deadline]
-					)),
-					accessList: [],
-				},
-			]
-			const results = await this.ethereum.multicall(swapTransactions, block.number + 1n)
-			if (results.length !== 3) throw ('invalid multicall result')
-			if (results[2].statusCode === 'success') {
-				const parsed = swapInterface.decodeFunctionResult('swapTokensForExactETH', results[2].returnValue)
-				const inOut = parsed.toObject() as { amounts: bigint[] } // TODO, change to funtype
-				if (inOut.amounts.length != 2) return []
-				if(inOut.amounts[0] <= 0n || inOut.amounts[1] <= 0n || token.decimals <= 0n) return []
-				inOutResults.push( {
-					token: addressString(token.token ),
-					inOutAmount: [inOut.amounts[1], inOut.amounts[0]],
-					decimals: token.decimals
-				})
+			const poolAddresses = calculateUniswapPools(token.address, CHAINS[chainString].weth)
+			const uniswapSpotCalls = getUniswapSpotCalls(token.address, quoteToken.address, poolAddresses)
+
+			const callData = stringToUint8Array(IMulticall3.encodeFunctionData('aggregate3', [uniswapSpotCalls]))
+			const callTransaction = {
+				type: '1559',
+				from: MOCK_ADDRESS,
+				to: MULTICALL3,
+				value: 0n,
+				input: callData,
+				maxFeePerGas: 0n,
+				maxPriorityFeePerGas: 0n,
+				gasLimit: 15_000_000n,
```

Comment:
This limit seems excessive, and it could result in unnecessary failures during multicalls with large simulation stacks.  Can we come up with an upper bound for this that is a bit more reasonable?  I don't know what a "normal" version of these calls costs, but maybe we can just take normal and multiply it by 2 or 3 or something?